src/vcpkg/build.cpp

```diff
@@ -895,17 +1080,22 @@ namespace vcpkg::Build
         auto stdoutlog = buildpath / ("stdout-" + action.spec.triplet().canonical_name() + ".log");
         int return_code;
         {
+            Sha512Scanner scanner;
             auto out_file = fs.open_for_write(stdoutlog, VCPKG_LINE_INFO);
             return_code = cmd_execute_and_stream_data(
                 command,
                 [&](StringView sv) {
                     print2(sv);
+                    if (action.build_options.auto_update_mismatched_sha512 == Build::AutoUpdateMismatchedSHA512::YES)
+                        scanner.parse_input(sv);
                     Checks::check_exit(VCPKG_LINE_INFO,
                                        out_file.write(sv.data(), 1, sv.size()) == sv.size(),
                                        "Error occurred while writing '%s'",
                                        stdoutlog);
                 },
                 env);
+            if (action.build_options.auto_update_mismatched_sha512 == Build::AutoUpdateMismatchedSHA512::YES)
+                scanner.replace_in_files(paths);
```

Comment:
Maybe do the replacement directly after the download and not when the whole port is done with building